addons/purchase/views/account_move_views.xml

```diff
@@ -33,6 +33,11 @@
             <xpath expr="//field[@name='line_ids']/tree/field[@name='company_id']" position="after">
                 <field name="purchase_line_id" invisible="1"/>
             </xpath>
+            <xpath expr="//div[@name='button_box']" position="inside">
+                <button class="oe_stat_button" name="action_view_source_purchase_orders" type="object" icon="fa-pencil-square-o" attrs="{'invisible': ['|', ('purchase_order_count', '=', 0), ('move_type', '!=', 'in_invoice')]}">
```

Comment:
What if it is a refund? (I'm not sure if there are x_order_ids in refunds, just asking)
Also, I am thinking why not showing this button when there are linked orders, regardless of being an in_invoice or not?

Reply:
I had to discuss this with CHKL, as the task only mentioned regular invoices and vendor bills and he agreed with you :slightly_smiling_face:  But I am still hiding it in journal entry views